pkg/k8s/informer/informer.go

```diff
@@ -36,37 +36,31 @@ func init() {
 	)
 }
 
-type ConvertFunc func(obj interface{}) interface{}
-
-// NewInformer is a copy of k8s.io/client-go/tools/cache/NewInformer with a new
-// argument which converts an object into another object that can be stored in
-// the local cache.
-func NewInformer(
+// NewTransformingInformer is a copy of k8s.io/client-go/tools/cache/NewTransformingInformer.
+func NewTransformingInformer(
```

Comment:
Given that this function is now a vanilla copy of the upstream one, I would suggest to update the callers to use the upstream one instead (either `NewInformer` or `NewTransformingInformer` depending on whether the transformer function is nil).

Reply:
Looks like `pkg/k8s/informer` does two things: it allows using a custom store and it injects a `CacheMutationDetector`. The custom store we can't do directly with the upstream informer, but the latter can be done with a transforming informer. Since we can't get rid of the custom store use yet, it's probably easier to leave the callers as is for now.

Reply:
Yeah, I was referring to `NewTransformingInformer` only, which doesn't take a custom store as parameter. But I missed that there is currently the custom `CacheMutatorDetector` part. Given that, I agree that it is probably better to not modify the callers for the moment (I would suggest to update the comment though, since it is currently a bit misleading).